smithyctl/command/component/package.go

```diff
@@ -0,0 +1,208 @@
+			if err := packageComponent(cmd.Context(), flags); err != nil {
+				return errors.Errorf("unexpected failure: %w", err)
+			}
+			return nil
+		},
+	}
+
+	cmd.
+		Flags().
+		StringVar(
+			&flags.specPath,
+			"spec-path",
+			".",
+			"the location of the component spec file",
+		)
+	cmd.
+		Flags().
+		StringVar(
+			&flags.packageVersion,
+			"version",
+			"",
+			"the version to be used to package the component",
+		)
+	_ = cmd.MarkFlagRequired("version")
+	cmd.
+		Flags().
+		StringVar(
+			&flags.sdkVersion,
+			"sdk-version",
+			"",
+			"the version the sdk used to build the component",
+		)
+	_ = cmd.MarkFlagRequired("sdk-version")
+	cmd.
+		Flags().
+		StringVar(
+			&flags.registryURL,
+			"registry-url",
+			"ghcr.io",
+			"the reference to the OCI compliant registry",
+		)
+	cmd.
+		Flags().
+		StringVar(
+			&flags.registryBaseRepository,
+			"registry-base-repository",
+			"smithy-security/manifests/components",
+			"the repository where to push manifests to",
+		)
+	cmd.
+		Flags().
+		BoolVar(
+			&flags.registryAuthEnabled,
+			"registry-auth-enabled",
+			false,
+			"if enabled, it requires authentication for the registry",
+		)
+	cmd.
+		Flags().
+		StringVar(
+			&flags.registryAuthUsername,
+			"registry-auth-username",
+			"",
+			"the username used for authentication for the registry",
+		)
+	cmd.
+		Flags().
+		StringVar(
+			&flags.registryAuthPassword,
+			"registry-auth-password",
+			"",
+			"the password used for authentication for the registry",
+		)
+
+	return cmd
+}
+
+func packageComponent(ctx context.Context, flags packageFlags) error {
+	component, err := parseComponentSpec(flags.specPath)
+	if err != nil {
+		return err
+	}
+
+	reg, err := registry.New(
+		flags.registryURL,
+		flags.registryBaseRepository,
+		flags.registryAuthEnabled,
+		flags.registryAuthUsername,
+		flags.registryAuthPassword,
+	)
+	if err != nil {
+		return errors.Errorf("failed to initialize registry: %w", err)
+	}
+
+	if err := reg.Package(ctx, registry.PackageRequest{
+		Component:        component,
+		SDKVersion:       flags.sdkVersion,
+		ComponentVersion: flags.packageVersion,
+	}); err != nil {
+		return errors.Errorf("failed to package component: %w", err)
+	}
+
+	return nil
+}
+
+func parseComponentSpec(path string) (*v1.Component, error) {
+	const (
+		defaultSmithyComponentFileNameYaml = "smithy-component.yaml"
+		defaultSmithyComponentFileNameYml  = "smithy-component.yml"
+	)
+
+	// If the path is not defined we look in the current directory.
+	if path == "" {
+		path = "."
+	}
+
+	// If the path doesn't exist, we return.
+	info, err := os.Stat(path)
+	if err != nil {
+		if os.IsNotExist(err) {
+			return nil, errors.New(fmt.Sprintf("%s does not exist", path))
+		}
+		return nil, fmt.Errorf("failed check config file: %w", err)
+	}
+
+	var fullPath = path
+	// If the path is a directory we check if it contains a smithy-component.yaml or smithy-component.yml.
+	if info.IsDir() {
+		files, err := os.ReadDir(path)
```

Comment:
this could be simplified to `is this a directory? if yes append component.yaml to it and call myself in a recursion for the new path`

Reply:
I ended up using the file directly